### PR TITLE
op-challenger: Introduce TraceAccessor

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

The game solver uses TraceAccessor instead of TraceProvider directly so that it has enough information to encapsulate a split trace provider game type.  Currently the only version of `TraceAccessor` used has a single trace provider, but the logic for calling a function to select the appropriate provider is implemented and tested in preparation for adding this plot game provider in a follow up PR.

Split out of https://github.com/ethereum-optimism/optimism/pull/8012/ and polished up.

**Tests**

Existing tests confirm GameSolver logic has not changed.  Unit tests added to ensure the provider selection function is used correctly.

**Metadata**

- https://github.com/ethereum-optimism/client-pod/issues/43

### CODE REVIEW

File: op-challenger/game/fault/types/types.go [hunk 1]
```diff
@@ -57,6 +57,11 @@ type OracleUpdater interface {
 	UpdateOracle(ctx context.Context, data *PreimageOracleData) error
 }
 
+type TraceAccessor interface {
```
Comment (Inphi) [inline op-challenger/game/fault/types/types.go:60]: nit: add some documentation.

File: op-challenger/game/fault/trace/select.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package trace
```
Comment (Inphi) [inline op-challenger/game/fault/trace/select.go:1]: bikeshed nit: This filename is a bit surprising. Was expecting something `access` related.

**Review by Inphi — APPROVED** [review]
The abstractions are tight!

File: op-challenger/game/fault/trace/select.go (hunk 2 above)
Comment (ajsutton) [reply op-challenger/game/fault/trace/select.go:1]: Yeah good spot - it made more sense earlier in the refactoring process. :)

**Pull request merged**